api_pool.go

```diff
@@ -0,0 +1,524 @@
+// List of current stake pools delegators.
+func (c *apiClient) PoolDelegator(
+	ctx context.Context,
+	poolID string,
+	query APIPagingParams,
+) ([]PoolDelegator, error) {
+	requestUrl, err := url.Parse(
+		fmt.Sprintf("%s/%s/%s/%s", c.server, resourcePool, poolID, resourcePoolDelegator),
+	)
+	if err != nil {
+		return []PoolDelegator{}, err
+	}
+
+	v := url.Values{}
+	if query.Count > 0 {
+		v.Set("count", fmt.Sprintf("%d", query.Count))
+		requestUrl.RawQuery = v.Encode()
+	}
+	if query.Page > 0 {
+		v.Set("page", fmt.Sprintf("%d", query.Page))
+		requestUrl.RawQuery = v.Encode()
+	}
+	if query.Order != "" {
+		v.Set("order", query.Order)
+		requestUrl.RawQuery = v.Encode()
+	}
+
+	req, err := http.NewRequest(http.MethodGet, requestUrl.String(), nil)
+	if err != nil {
+		return []PoolDelegator{}, err
+	}
+
+	req.Header.Add("project_id", c.projectId)
+	req = req.WithContext(ctx)
+
+	res, err := c.client.Do(req)
+	if err != nil {
+		return []PoolDelegator{}, err
+	}
+	defer res.Body.Close()
+
+	if res.StatusCode != http.StatusOK {
+		return []PoolDelegator{}, handleAPIErrorResponse(res)
+	}
+	pool := []PoolDelegator{}
+	err = json.NewDecoder(res.Body).Decode(&pool)
+	if err != nil {
+		return []PoolDelegator{}, err
+	}
+	return pool, nil
+}
+
+// PoolBlocks returns the Stake pool blocks
+// List of stake pools blocks.
+func (c *apiClient) PoolBlocks(
+	ctx context.Context,
+	poolID string,
+	query APIPagingParams,
+) (PoolBlocks, error) {
+	requestUrl, err := url.Parse(
+		fmt.Sprintf("%s/%s/%s/%s", c.server, resourcePool, poolID, resourcePoolBlocks),
+	)
+	if err != nil {
+		return PoolBlocks{}, err
+	}
+
+	v := url.Values{}
+	if query.Count > 0 {
+		v.Set("count", fmt.Sprintf("%d", query.Count))
+		requestUrl.RawQuery = v.Encode()
+	}
+	if query.Page > 0 {
+		v.Set("page", fmt.Sprintf("%d", query.Page))
+		requestUrl.RawQuery = v.Encode()
+	}
+	if query.Order != "" {
+		v.Set("order", query.Order)
+		requestUrl.RawQuery = v.Encode()
+	}
+
+	req, err := http.NewRequest(http.MethodGet, requestUrl.String(), nil)
+	if err != nil {
+		return PoolBlocks{}, err
+	}
+
+	req.Header.Add("project_id", c.projectId)
+	req = req.WithContext(ctx)
+
+	res, err := c.client.Do(req)
+	if err != nil {
+		return PoolBlocks{}, err
+	}
+	defer res.Body.Close()
+
+	if res.StatusCode != http.StatusOK {
+		return PoolBlocks{}, handleAPIErrorResponse(res)
+	}
+	pool := PoolBlocks{}
+	err = json.NewDecoder(res.Body).Decode(&pool)
+	if err != nil {
+		return PoolBlocks{}, err
+	}
+	return pool, nil
+}
+
+// PoolUpdate returns the Stake pool updates
+// List of certificate updates to the stake pool.
+func (c *apiClient) PoolUpdate(
+	ctx context.Context,
+	poolID string,
+	query APIPagingParams,
+) ([]PoolUpdate, error) {
+	requestUrl, err := url.Parse(
+		fmt.Sprintf("%s/%s/%s/%s", c.server, resourcePool, poolID, resourcePoolUpdate),
+	)
+	if err != nil {
+		return []PoolUpdate{}, err
+	}
+
+	v := url.Values{}
+	if query.Count > 0 {
```

Comment:
query validation and formatting moved to `formatParams(v url.Values, query APIPagingParams)`. Implementation of function in util.go

Reply:
OK